models/news.models.js

```diff
@@ -28,17 +28,23 @@ exports.fetchArticleByid=(id)=>{
 }
 
 exports.fetchAllArticles=async()=>{
-    response=await db.query(`SELECT * FROM articles ORDER BY created_at desc`)
+    const response=await db.query(`SELECT * FROM articles ORDER BY created_at desc`)
         const articles=response.rows;
-        preparedarticles=articles.map(async(article)=>{
-            const newarticle={...article}
-            countResult=await db.query (`SELECT COUNT(body) FROM comments WHERE article_id=${newarticle.article_id}`)
-            newarticle.comment_count=countResult.rows[0].count
-            delete newarticle.body
-            return newarticle;
+        const preparedArticles=articles.map(async(article)=>{
+            const newArticle={...article}
+            const countResult=await db.query (`SELECT COUNT(body) FROM comments WHERE article_id=${newArticle.article_id}`)
+            newArticle.comment_count=countResult.rows[0].count
+            delete newArticle.body
+            return newArticle;
         })
-        const formattedarticles=await Promise.all(preparedarticles)
-        return formattedarticles;
+        const formattedArticles=await Promise.all(preparedArticles)
+        return formattedArticles;
```

Comment:
One thing I missed when I looked at your previous work was that this doesn't need to be so manual. You're manually going into each article and requesting the comments and then reformatting that to add the comment_count property but this is poor use of PSQL.

We can make **one** psql statement that does the select, grabs the comments, aggregates them and puts them in a column called comment_count, and finally orders the response in descending order. **In a separate branch** please work on this statement and refactor it to do all the work in one statement. It is possible, remember we have JOINS available to us in SQL.